### DISCUSSION TITLE
Subdomains part of Pockethost

### DISCUSSION THREAD

**Discussion by pragmatech** [pockethost/pockethost#91] [Q&A]
If not asking too much: What service, project, etc. is the basis for the platform-y subdomains-part of Pockethost?
thanks,
Darren

**Comment by benallfree** [comment]
Hi Darren, 

> What service, project, etc. is the basis for the platform-y subdomains-part of Pockethost?

I'm struggling to parse that question - are you asking why all the instances are `<something>.pockethost.io`?

pockethost.io is the domain where we are hosting the service. It has a wildcard SSL cert which covers all the subdomains. #25 will add custom domain support at some point.

I hope that answers your question! If not, feel free to expand please.

**Reply by pragmatech** [reply]
Thanks for your reply. Sorry if I was unclear. 
To the extent of my fairly naive web dev chops, I've messed with stuff like https://github.com/vercel/platforms to learn about a 'Multi-Tenant App with Custom Domains'. The subdomain part of the puzzle obviously seems key to, say, how wordpress.com becomes widely adopted (ignoring the prerequisite 'killer app' that WP ... was). 
Put another way – in my break-it-down-to-simpler approaches personal education (the Nextjs, Vercel magic gets hard to grok for me) When I look at Pockethost, I'm wanting to know what the 'Hello World' of the multi-tenant facility alone sort of looks like. (I may need to just slog through and not be lazy!) 
Last way ... Wow! Darren purchased 'human.me' [he didn't] and he's sure his little app will be attractive as john.human.me, mary.human.me ... and approximatley 1/5 of the entire planet! (he's optomistic). Where might he go to get savy on multi-tenancy as so nicely implemented as Ben's?
Thanks again for sharing your stuff,
Darren

**Comment by benallfree** [comment]
It sounds like you are trying to use PocketBase to build a multitenant app for your users. Is that correct?

There are lots of considerations for multitenancy. As you can see from this repo, it's not easy to set up. If you're trying to build your own multitenant app using PocketBase, there may be some things in this repo to help you, although the multitenancy we cover here is specific to allowing each user their own copy of PocketBase,

**Reply by pragmatech** [reply]
Sorry, I should have more clearly stated that I'm just trying to get it together re: a more generic usage of how wildcard subdomains and multitenancy works. As such, apologies if such a query is too out of the scope of Pocketbase/host. To be clear, my very hypothetical usage of such subdomaining is more in a localized community (commons) thing. 
thanks!
ps- your 'Local Domain Setup Instructions' seems to get to the aspect i was wondering about. I should spend more time exploring the codebase.

**Reply by benallfree** [reply]
Sounds good. I'm hearing that you're curious about building a multitenant/subdomain app powered by one instance of PocketBase underneath. 

If so, exploring this code base *might* help, but then again it might not. The purpose of PocketHost is to run a separate instance of PocketBase on each subdomain so developers don't need to host their own instances of PocketBase. The PocketHost frontend is NOT multitenant. In other words, browsing to different subdomains on PocketHost will activate separate instances of PocketBase rather than refer to a single instance of PocketBase.
